File: app/src/main/java/mb00/android/codehub/ui/search/inject/SearchCodeFragmentModule.java
package mb00.android.codehub.ui.search.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.search.interactor.SearchInteractor;
import mb00.android.codehub.ui.search.router.SearchRouter;
import mb00.android.codehub.ui.search.viewmodel.SearchCodeViewModel;


@Module
public class SearchCodeFragmentModule {

    @Provides
    SearchCodeViewModel provideSearchCodeViewModel(SearchInteractor interactor, SearchRouter router) {
        return new SearchCodeViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/search/inject/SearchCodeFragmentModule.kt
package mb00.android.codehub.ui.search.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.search.interactor.SearchInteractor
import mb00.android.codehub.ui.search.router.SearchRouter
import mb00.android.codehub.ui.search.viewmodel.SearchCodeViewModel


@Module
class SearchCodeFragmentModule {

    @Provides
    internal fun provideSearchCodeViewModel(interactor: SearchInteractor, router: SearchRouter): SearchCodeViewModel {
        return SearchCodeViewModel(interactor, router)
    }

}
